plugins/poll/app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DiscoursePoll::PollsController < ::ApplicationController
+  requires_plugin DiscoursePoll::PLUGIN_NAME
+
+  before_action :ensure_logged_in, except: [:voters, :grouped_poll_results]
+
+  def vote
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+    options = params.require(:options)
+
+    begin
+      poll, options = DiscoursePoll::Poll.vote(post_id, poll_name, options, current_user)
+      render json: { poll: poll, vote: options }
```

Comment:
Why is `vote` called `options`? 🤔

Reply:
I am not sure exactly why, but this is the name in JavaScript too. It makes sense because a vote is nothing more than a set of selected poll options.